Add doc to setup continuous integration E2E tests for extensions (#17537)
docker/docs: desktop/extensions-sdk/dev/continuous-integration.md

---
title: "Continuous Integration"
description: Automatically test and validate your extension.
keywords: Docker, Extensions, sdk, CI, test, regression
---

> In order to help validating your extension and ensure it's functional, Docker Desktop provides a set of tools to help you set continuous integration for your extension.

Reviewer: Maybe we should mention that they are experimental, as we did in the package readme. WDYT @gtardif?

## Setup CI environment with Github action

You need Docker Desktop to be able to install and validate your extension.
You can start Docker Desktop in Github Actions using the [Docker Desktop Action](https://github.com/docker/desktop-action), with the following step:

```yaml
steps:
  - id: start_desktop
    uses: docker/desktop-action/start@v0.1.0
```

> > This action supports only Github Action Mac nodes at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.

Reviewer: In the GHA doc they call nodes runners: https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#runners

```suggestion
> This action supports only Github Action macOS runners at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.
```

---

> title: "Continuous Integration"

Reviewer 2: I know it's obvious in this context but just so we confirm to style and unpack the acronym, how about…

```suggestion
title: "Continuous Integration (CI)"
```

description: Automatically test and validate your extension.
keywords: Docker, Extensions, sdk, CI, test, regression
---

In order to help validating your extension and ensure it's functional, the Extension SDK provides a set of tools to help you set continuous integration for your extension.

> > The github action and the extension-test-helper library are both [experimental](https://docs.docker.com/release-lifecycle/#experimental).

Reviewer 2: Is there a link for more info on the extension-test-helper library?

Author: Good catch, I added both links here (the GHA link was already mentioned below)

{: .important }

## Setup CI environment with Github action

You need Docker Desktop to be able to install and validate your extension.
You can start Docker Desktop in Github Actions using the [Docker Desktop Action](https://github.com/docker/desktop-action), with the following step:

```yaml
steps:
  - id: start_desktop
    uses: docker/desktop-action/start@v0.1.0
```

> > This action supports only Github Action macOS runners at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.

Reviewer 2: ```suggestion
> This action supports only Github Action macOS runners at the moment. You need to specify `runs-on: macOS-latest` for your end to end tests.
```

{: .important }

> Once the step has executed, the next steps can use Docker Desktop and the Docker CLI to install an test the extension

Reviewer 2: ```suggestion
Once the step has executed, the next steps can use Docker Desktop and the Docker CLI to install and test the extension.
```

## Validating your extension with puppeteer

> Once Docker Desktop is started in the CI, you can build, install and validate your extension with jest and puppeteer.

Reviewer 2: ```suggestion
Once Docker Desktop is started in the CI, you can build, install, and validate your extension with jest and puppeteer.
```
